### PR TITLE
[feat]: modify sample weight & add pairwise loss to multi tower model

### PR DESCRIPTION
**Pull request by yangxudong**
1. remove dependency of the fix position of sample weight column
2. fix bug of share embedding of id feature and sequence feature
3. add pairwise loss option to multi tower model

### CODE REVIEW

**Review by dawn310826 — COMMENTED** [review]
add test and docs

File: easy_rec/python/compat/feature_column/feature_column_v2.py [hunk 1]
```diff
@@ -1056,12 +1056,12 @@ def model_fn(features, ...):
...
-          'To use shared_embedding_column, all categorical_columns must have '
-          'the same type, or be weighted_categorical_column of the same type. '
-          'Given column: {} of type: {} does not match given column: {} of '
-          'type: {}'.format(c0, type(c0), c, type(c)))
+    # if not isinstance(c, type(c0)):
+    #   raise ValueError(
+    #       'To use shared_embedding_column, all categorical_columns must have '
+    #       'the same type, or be weighted_categorical_column of the same type. '
+    #       'Given column: {} of type: {} does not match given column: {} of '
+    #       'type: {}'.format(c0, type(c0), c, type(c)))
```
Comment (paradiseHIT) [inline easy_rec/python/compat/feature_column/feature_column_v2.py:1064]: 这里注释直接删掉是不是就好了？

另外，后面还是建议一个feature一个pull request

File: easy_rec/python/model/multi_tower.py [hunk 2]
```diff
@@ -22,6 +23,8 @@ def __init__(self,
                is_training=False):
     super(MultiTower, self).__init__(model_config, feature_configs, features,
                                      labels, is_training)
+    self._losses = self._model_config.losses
+    print("loss num:", len(self._losses))
```
Comment (paradiseHIT) [inline easy_rec/python/model/multi_tower.py:27]: 这里是否应该调用标准的log函数，logging.info？

**Review by paradiseHIT — COMMENTED** [review]
没什么大的问题，小问题单独comment了

File: easy_rec/python/compat/feature_column/feature_column_v2.py (hunk 1 above)
Comment (yangxudong) [reply easy_rec/python/compat/feature_column/feature_column_v2.py]: 已修改

File: easy_rec/python/model/multi_tower.py (hunk 2 above)
Comment (yangxudong) [reply easy_rec/python/model/multi_tower.py]: 已修改

**Comment by yangxudong** [comment]
> 

已补充

**Comment by yangxudong** [comment]
> add test and docs

已添加

File: easy_rec/python/input/input.py [hunk 3]
```diff
@@ -194,20 +195,23 @@ def create_placeholders(self, export_config):
     input_vals = tf.string_split(
         inputs_placeholder, self._data_config.separator,
         skip_empty=False).values
+
+    sample_weight_field = ""
+    if self._data_config.HasField('sample_weight'):
```
Comment (poson) [inline easy_rec/python/input/input.py:200]: 159和199行，这三行重复了

File: easy_rec/python/model/multi_tower.py [hunk 4]
```diff
@@ -60,3 +64,21 @@ def build_predict_graph(self):
...
+    for loss in self._losses:
+      if loss.loss_type == LossType.PAIR_WISE_LOSS:
+        loss_value = pairwise_loss(labels, logits)
+        self._loss_dict['pairwise_loss'] = loss_value * loss.weight
+      elif loss.loss_type == LossType.CLASSIFICATION:
+        loss_value = tf.losses.sigmoid_cross_entropy(labels, logits,
+                                                     self._sample_weight)
+        self._loss_dict['sigmoid_loss'] = loss_value * loss.weight
+      else:
+        logging.warning("Unsupported loss type: %s" % loss.loss_type)
```
Comment (poson) [inline easy_rec/python/model/multi_tower.py:83]: 这个应该raise 异常？

File: easy_rec/python/input/input.py (hunk 3 above)
Comment (yangxudong) [reply easy_rec/python/input/input.py:200]: 这个是在两个不同的函数里面，分别对应了两种情况下的执行，都需要改的

File: easy_rec/python/model/multi_tower.py (hunk 4 above)
Comment (yangxudong) [reply easy_rec/python/model/multi_tower.py]: 已添加

**Review by poson — APPROVED** [review]
